sql/src/test/java/io/crate/integrationtests/disruption/seqno/SimpleVersioningIT.java

```diff
@@ -0,0 +1,132 @@
+
+public class SimpleVersioningIT extends SQLTransportIntegrationTest {
+
+    @Test
+    public void test_compare_and_set() {
+
+        execute("create table test (id integer primary key, value string)");
+        ensureGreen();
+
+        execute("insert into test (id, value) values (?, ?) returning _seq_no, _primary_term", new Object[]{1, "value1_1"});
+
+        long seqNo = (long) response.rows()[0][0];
+        long primaryTerm = (long) response.rows()[0][1];
+
+        assertThat(seqNo, equalTo(0L));
+        assertThat(primaryTerm, equalTo(1L));
+
+        execute("update test set value = 'value1_2' where id = 1 and _seq_no = 0 and _primary_term = 1 returning _seq_no, _primary_term");
+
+        seqNo = (long) response.rows()[0][0];
+        primaryTerm = (long) response.rows()[0][1];
+
+        assertThat(seqNo, equalTo(1L));
+        assertThat(primaryTerm, equalTo(1L));
+
+
+        execute("update test set value = 'value1_1' where id = 1 and _seq_no = 10 and _primary_term = 1 returning _seq_no, _primary_term");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("update test set value = 'value1_1' where id = 1 and _seq_no = 10 and _primary_term = 2 returning _seq_no, _primary_term");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("update test set value = 'value1_1' where id = 1 and _seq_no = 1 and _primary_term = 2 returning _seq_no, _primary_term");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 10 and _primary_term = 1");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 10 and _primary_term = 2");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("delete from test where id = 1 and _seq_no = 1 and _primary_term = 2");
+        assertThat(response.rowCount(), equalTo(0L));
+
+        execute("refresh table test");
+
+        for (int i = 0; i < 10; i++) {
+            execute("select _seq_no, _primary_term from test where id = 1");
+            assertThat(response.rows()[0][0], equalTo(1L));
+            assertThat(response.rows()[0][1], equalTo(1L));
+        }
+
+        // select with versioning
+        for (int i = 0; i < 10; i++) {
+            execute("select _seq_no, _primary_term, _version from test");
+            assertThat(response.rows()[0][0], equalTo(1L));
+            assertThat(response.rows()[0][1], equalTo(1L));
+            assertThat(response.rows()[0][2], equalTo(2L));
+        }
+
+        // select without versioning
+        for (int i = 0; i < 10; i++) {
+            execute("select _seq_no, _primary_term from test");
+            assertThat(response.rows()[0][0], equalTo(1L));
+            assertThat(response.rows()[0][1], equalTo(1L));
+        }
+
+        execute("delete from test where id = 1 and _seq_no = 1 and _primary_term = 1");
+        assertThat(response.rowCount(), equalTo(1L));
+
```

Comment:
This assertion is missing but would be good to have I think https://github.com/elastic/elasticsearch/blob/master/server/src/test/java/org/elasticsearch/versioning/SimpleVersioningIT.java#L286

Reply:
The reason why i left it out is, because the row is already deleted here.

Reply:
the additional assertions should validate that the row is already deleted in any case, see es code

Reply:
The assertion from es validates that the `VersionConflictEngineException` is still thrown even that it is deleted. we don't throw a `VersionConflictEngineException` at all and we validated already that it is deleted it. However, i can add it.